includes/payment-methods/class-wc-stripe-express-checkout-ajax-handler.php

```
@@ -262,7 +276,19 @@ public function ajax_get_selected_product_data() {
 
 		try {
 			$product_id      = isset( $_POST['product_id'] ) ? absint( $_POST['product_id'] ) : 0;
-			$qty             = ! isset( $_POST['qty'] ) ? 1 : apply_filters( 'woocommerce_add_to_cart_quantity', absint( $_POST['qty'] ), $product_id );
+			// Preserve decimal quantities (see ajax_add_to_cart above) by parsing with
+			// wc_stock_amount() instead of absint() before applying the core filter.
+			// Reject array quantities (e.g. qty[]=1) and zero/negative values before they
+			// flow into wc_stock_amount() and the price math below.
+			if ( isset( $_POST['qty'] ) && is_array( $_POST['qty'] ) ) {
+				throw new Exception( __( 'Invalid product quantity.', 'woocommerce-gateway-stripe' ) );
```

PHPMD (2.15.0)

[error] Missing class import via use statement (line '284', column '15'). (undefined)

(MissingImport)